webapp.py

```diff
@@ -0,0 +1,26 @@
+from flask import Flask, render_template, request, jsonify
+from chatbot import message_to_bot, setup
+
+app = Flask(__name__)
+
+
+@app.route("/")
+def index():
+    return render_template("index.html")
+
+
+@app.route("/process", methods=["POST"])
+def process():
+    global clf, learn_response
+
+    user_input = request.form["usrInput"]
+    if user_input:
+        bot_text, learn_response = message_to_bot(user_input, clf, learn_response)
+        return jsonify({"botText": bot_text})
+
+    return jsonify({"error": "Missing data!"})
+
+
+if __name__ == "__main__":
+    clf, learn_response = setup()
+    app.run(debug=True)
```

Comment:
I see that we do have the `app.py` file already, can we integrate this code there?

Reply:
`app.py` is a bit buggy and was also changed during the SlackBot PR. We can surely integrate it into one program but I didn't want to mess things up by merge conflicts. It needed a bit more effort integrating both files and making sure every case is handled correctly. Hence, the different file. Also, `app.py` keeps the `user_input` in the URL and uses a GET request, _and the Slack Bot depends on it_. Whereas, in `webapp.py` the `user_input` is handled with a POST request, making it cleaner.

> GET carries request parameter appended in URL string while POST carries request parameter in message body which makes it more secure way of transferring data from client to server in HTTP protocol.

Refs: [[1]](https://www.w3schools.com/tags/ref_httpmethods.asp), [[2]](https://www.tutorialspoint.com/listtutorial/Difference-between-GET-and-POST-method-in-HTTP/3916)

> We would, however, need to add running this in the readme.

Sure. Would you want `webapp.py` to be added as an alternative to `init.py` or as a replacement?

Reply:
Hmm I see. Let's just lease it the way it is for now then. 

> Sure. Would you want `webapp.py` to be added as an alternative to `init.py` or as a replacement, @vishakha-lall?

An alternative, as a subheading 'Setting up UI'